### PR TITLE
[receiver/memcached] Fix some issues in metadata.yaml

### LINKED ISSUE #6881: [receiver/memcached] Units need revisiting
Particularly, [memcached.cpu.usage](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/4fedae5d026f52dc0cbf26b2a619e7105c0eac9e/receiver/memcachedreceiver/metadata.yaml#L95) appears to be wrong. This should be a unit of time.

Most of the other units should use the `"{things}"` notation.

### LINKED ISSUE #6886: [receiver/memcached] Revisit metric types
The following metrics reported as double values but exposed as int stats by memcached: 
- `threads` memcached stat -> `memcached.threads` metric (Number of threads used by the memcached instance) 
- `curr_items` memcached stat -> `memcached.current_items` metric (Number of items currently stored in the cache)

### PR DESCRIPTION
**Pull request by djaglowski**
Resolves #6881

Resolves #6886

### CODE REVIEW

File: receiver/memcachedreceiver/metadata.yaml [hunk 1]
```diff
@@ -101,17 +101,17 @@ metrics:
...
       monotonic: true
       aggregation: cumulative
     attributes: [state]
   memcached.threads:
     description: Number of threads used by the memcached instance.
-    unit: 1
+    unit: "{threads}"
     sum:
-      value_type: double
+      value_type: int
```
Comment (dmitryax) [inline receiver/memcachedreceiver/metadata.yaml:114]: Thank! Can you please do the same for `memcached.current_items`?

**Review by dmitryax — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Please resolve the conflicts.

**Comment by bogdandrutu** [comment]
@djaglowski try one more time :)

**Comment by dmitryax** [comment]
Sorry @djaglowski :)

**Pull request merged**